nanoFramework.WebServer/WebServer.cs

```diff
@@ -537,35 +534,69 @@ private void StartListener()
                         // Variables used only within the "for". They are here for performance reasons
                         bool mustAuthenticate;
                         bool isAuthOk;
-                        bool isRoute = false;
+                        CallbackRoutes selectedRoute = null;
+                        string multipleCallback = null;
 
                         foreach (CallbackRoutes route in _callbackRoutes)
                         {
                             if (!IsRouteMatch(route, context.Request.HttpMethod, context.Request.RawUrl))
                             {
                                 continue;
                             }
+                            if (selectedRoute is null)
+                            {
+                                selectedRoute = route;
+                            }
+                            else
+                            {
+                                multipleCallback ??= $"Multiple matching callbacks: {selectedRoute.Callback.DeclaringType.FullName}.{selectedRoute.Callback.Name}";
+                                multipleCallback += $", {route.Callback.DeclaringType.FullName}.{route.Callback.Name}";
+                            }
+                        }
 
-                            isRoute = true;
+                        if (multipleCallback is not null)
+                        {
+                            multipleCallback += ".";
+                            context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                            OutPutStream(context.Response, multipleCallback);
 
+                            HandleContextResponse(context);
+                        }
+                        else if (selectedRoute is null)
+                        {
+                            if (CommandReceived != null)
```

Comment:
You still need to check auth. So, you should have to adjust with the previous logic.